src/Imports/ModelImporter.php

```diff
@@ -56,12 +64,13 @@ public function import(Worksheet $worksheet, ToModel $import, int $startRow = 1)
 
         $this->manager->setRemembersRowNumber(method_exists($import, 'rememberRowNumber'));
 
-        $i = 0;
+        $i             = 0;
+        $batchStartRow = $startRow;
         foreach ($worksheet->getRowIterator($startRow, $endRow) as $spreadSheetRow) {
             $i++;
 
             $row = new Row($spreadSheetRow, $headingRow, $headerIsGrouped);
-            if (!$import instanceof SkipsEmptyRows || ($import instanceof SkipsEmptyRows && !$row->isEmpty($withCalcFormulas))) {
+            if (!$import instanceof SkipsEmptyRows || !$row->isEmpty($withCalcFormulas)) {
```

Comment:
isEmpty should only be called if SkipsEmptyRows concern is present; would be breaking change to remove

Reply:
It's not, you already have `!$import instanceof SkipsEmptyRows` so it's a redundant condition, that's all